src/patterns/gen-ai/aws-text-to-sql/index.ts

```diff
@@ -0,0 +1,1219 @@
+  } // end construct
+
+  private validateDbProps(props: TextToSqlProps): void {
+    // Check if existingAuroraDbCluster and databaseClusterProps are set at the same time
+    if (props.existingAuroraDbCluster && props.databaseClusterProps) {
+      throw new Error(
+        'Only one of existingAuroraDbCluster or databaseClusterProps can be set at a time.',
+      );
+    }
+
+    // Check if databaseInstanceProps and existingRdsDbInstance are set at the same time
+    if (props.databaseInstanceProps && props.existingRdsDbInstance) {
+      throw new Error(
+        'Only one of databaseInstanceProps or existingRdsDbInstance can be set at a time.',
+      );
+    }
+
+    // Check if existingAuroraDbCluster and existingRdsDbInstance are set at the same time
+    if (props.existingAuroraDbCluster && props.existingRdsDbInstance) {
+      throw new Error(
+        'Only one of existingAuroraDbCluster or existingRdsDbInstance can be set at a time.',
+      );
+    }
+  }
+
+  private createSecret(): secretsmanager.Secret {
+    return new secretsmanager.Secret(this, 'AuroraSecret' + this.stage, {
+      secretName: 'texttosqldbsecret',
+      generateSecretString: {
+        excludePunctuation: true,
+        passwordLength: 16,
+        secretStringTemplate: JSON.stringify({
+          username: 'admin',
+        }),
+        generateStringKey: 'password',
+      },
+    });
+  }
+
+  private createRdsInstance(
+    props: TextToSqlProps,
+    dbPort: number,
+  ): rds.DatabaseInstance {
+    switch (props.dbName) {
+      case DbName.MYSQL:
+        const instanceIdentifier = 'mysql-01';
+        return new rds.DatabaseInstance(this, 'MysqlRdsInstance', {
+          vpcSubnets: {
+            onePerAz: true,
+            subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
+          },
+          securityGroups: [this.dbSecurityGroup],
+          credentials: rds.Credentials.fromSecret(this.secret),
+          vpc: this.vpc,
+          port: dbPort,
+          databaseName: 'textosqldb' + this.stage,
+          allocatedStorage: 20,
+          instanceIdentifier,
+          engine: rds.DatabaseInstanceEngine.mysql({
+            version: rds.MysqlEngineVersion.VER_8_0_37,
+          }),
+          instanceType: ec2.InstanceType.of(
+            ec2.InstanceClass.T2,
+            ec2.InstanceSize.LARGE,
+          ),
+        });
+      // case DbName.POSTGRESQL:
+      //   // Add code for RDS PostgreSQL instance
+      //   return undefined;
+      default:
+        throw new Error('Invalid database name');
+    }
+  }
+
+  private createAuroraCluster(
+    props: TextToSqlProps,
+    dbPort: number,
+  ): rds.DatabaseCluster {
+    switch (props.dbName) {
+      case DbName.MYSQL:
+        return new rds.DatabaseCluster(this, 'AuroraCluster', {
+          engine: rds.DatabaseClusterEngine.auroraMysql({
+            version: rds.AuroraMysqlEngineVersion.VER_3_07_0,
+          }),
+          port: dbPort,
+
```

Comment:
is it missing credentials: rds.Credentials.fromSecret(this.secret), ?

Reply:
d28bbe26726f810317c2b3533f33c248fd7c9cfb